### PR TITLE
H5Pset_layout updates due to HDF5 change in behavior.

### PR DESCRIPTION
**Pull request by brtnfld**
As of HDF5 2.0, H5Pset* routines fail when used on default property lists.

### CODE REVIEW

File: .github/workflows/ubuntu_mpich.yml
Comment (jhendersonHDF) [inline .github/workflows/ubuntu_mpich.yml:1]: What's the reason for going back to disabling the I/O test?

Comment (brtnfld) [reply .github/workflows/ubuntu_mpich.yml:1]: I was going to address the iotest failing in another PR.

Comment (brtnfld) [reply .github/workflows/ubuntu_mpich.yml:1]: I'll see if I can fix the failures as part of this PR.

Comment (brtnfld) [reply .github/workflows/ubuntu_mpich.yml:1]: @jhendersonHDF, I fixed the issue in hdf5-iotest

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**